packages/yew-router-macro/src/routable_derive.rs

```diff
@@ -176,7 +176,7 @@ impl Routable {
                     }
 
                     quote! {
-                        Self::#ident { #(#fields),* } => ::std::format!(#right, #(#fields = #fields),*)
+                        Self::#ident { #(#fields),* } => ::std::format!(#right, #(#fields = yew_router::encode_for_url(&format!("{}", #fields))),*)
```

Comment:
I have not got time to test this change, but I think this should not be applied on the remainder path matcher(*）which would otherwise escape the remainder path.

Reply:
I've just done a quick test and it seems to work correctly in its current state when the link is generated and clicked.

Is this what you were concerned about failing?

### Code
<img width="653" alt="Test Source Code" src="https://user-images.githubusercontent.com/40693089/228122514-ea462ade-0247-4b66-ab93-0f0a2ab3b0b2.png">

### Result
<img width="458" alt="Rendered Result" src="https://user-images.githubusercontent.com/40693089/228122523-c639c485-c8e2-48a6-911f-9e277a320aaa.png">

Reply:
In route recognizer, you can construct a route like `/external/*remainder` (referred to as "named wildcards" in their documentation) which it will match both `/external/a` and `/external/a/b/c/d` and collects `a` and `a/b/c/d` into the route parameter respectively.
If we apply escaping to this, then we will not be able to construct routes like `/external/a/b/c/d` with `Link`.

See: https://docs.rs/route-recognizer/latest/route_recognizer/